pkg/api/unversioned/types.go

```diff
@@ -54,6 +54,38 @@ type ListMeta struct {
 	ResourceVersion string `json:"resourceVersion,omitempty"`
 }
 
+// ListOptions is the query options to a standard REST list/watch calls.
+type ListOptions struct {
+	TypeMeta `json:",inline"`
+
+
+	// FIXME: It would be super convinient to have LabelSelector and FieldSelector
+	// of types: labels.Selector and fields.Selector respectively (no additional parsing
+	// in random places of the code).
+	//
+	// However, this would require (at least);
+	// - adding support for serialization/deserialization labels/fields.Selector to/from JSON
+	//   [my feeling is that MarshalJSON and UnmarshalJSON is what needs to be implemented for them]
+	// - adding some special treatment of those fields in convert to URL params - this would be
+  //   a dedicated, non-generic code (I think).
```

Comment:
Indentation on last line is wrong ;)

If you implement the text marshaller interfaces, will that work for both url.Value and json marshalling? I think it's probably possible to make it work without writing custom codec code.

@smarterclayton may want to argue for copying these to every group/version that uses them. I think we can do that when we have versions that start needing different options.